### PR TITLE
Feature list tool functions

### PR DESCRIPTION
**Pull request by rkausch**
Fixes VORTEX-5590

## Proposed Changes
  - Repackaging functions into statusbar and non-statusbar functions
  - Change factory to allow for new functions to be added by other plugins
  - Insert factory into AnalysisToolbox to allow other plugins to access and add functions
  - Add ability for a function to declare what datatypes it supports to limit function availability when not needed
  - Add ability for a function to specify a default output for when a function doesn't have enough data, or doesn't apply to the current selection.

### CODE REVIEW

File: open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java [hunk 1]
```diff
@@ -188,6 +191,84 @@ public static double toDouble(Number value, double defaultValue)
...
+     * Converts the supplied value to a {@link BigDecimal}.
+     *
+     * @param number the value to convert to a {@link BigDecimal}.
+     * @return the {@link BigDecimal} equivalent of the supplied value.
+     */
+    public static BigDecimal toBigDecimal(Number number)
+    {
+        if (number instanceof BigDecimal)
+        {
+            return (BigDecimal)number;
```
Comment (bcleveland1022) [inline open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java]: Can we get rid of the multiple returns.

File: open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java [hunk 2]
```diff
@@ -188,6 +191,84 @@ public static double toDouble(Number value, double defaultValue)
...
+     * {@link Double#NaN} or infinity.
+     *
+     * @param value the value to test.
+     * @return true if the supplied value is a special case, false otherwise.
+     */
+    public static boolean isSpecial(Number value)
+    {
+        if (value instanceof Double)
+        {
+            return Double.isNaN(value.doubleValue()) || Double.isInfinite(value.doubleValue());
```
Comment (bcleveland1022) [inline open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java]: Can we remove multiple returns.

File: open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java [hunk 3]
```diff
@@ -188,6 +191,84 @@ public static double toDouble(Number value, double defaultValue)
...
+     *         is is numerically equal to the right side, a value of 0 is
+     *         returned; a value less than 0 if left is numerically less than
+     *         right; and a value greater than 0 if left is numerically greater
+     *         than right.
+     */
+    public static int compare(Number left, Number right)
+    {
+        if (NumberUtilities.isSpecial(left) || NumberUtilities.isSpecial(right))
+        {
+            return Double.compare(left.doubleValue(), right.doubleValue());
```
Comment (bcleveland1022) [inline open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java]: multiple returns.

Comment (rkausch) [reply open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java]: fixed

File: open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java (hunk 2 above)
Comment (rkausch) [reply open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java]: fixed

File: open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java (hunk 1 above)
Comment (rkausch) [reply open-sphere-base/core/src/main/java/io/opensphere/core/util/lang/NumberUtilities.java]: fixed

**Pull request merged**